docs/specs/account-linking.md

```diff
@@ -0,0 +1,421 @@
+
+For detail about default behaviors, please read the [Default Behaviors](#default-behaviors) section.
+
+### Default Behaviors
+
+The account linking config will be read according to the below precedence:
+
+1. If exist, always use the configurations in `account_linking` inside the current flow object in `signup_flows`.
+2. Else, use the `authentication_flow.default_account_linking` confuguration, if exist.
+3. Else, it is the built in default behavior. Please read the following sections for detail.
+
+#### The Current Defaults
+
+The current defaults are identical to the following config:
+
+```yaml
+oauth_claim:
+  pointer: "/email"
+user_profile_attribute:
+  pointer: "/email"
+action: error
+```
+
+These defaults will be applied to all oauth providers if account linking is not configured for that provider.
+
+Please refer to [Defining Linkings](#defining-linkings) and [Linking Actions](#linking-actions) sections for the meanings of the configs.
+
+#### The default linking of different provider types
+
+Different default account linking configs could be provided for different type of providers. However, this is not implemented at the moment.
+
+## Identity Attributes
+
+In authgear, each identity contributes to some standard attributes of the authgear user profile. Please see the following table for the standard attributes contributed by the email, username and phone identities.
+
+| Identity Type | Value            | Standard Attributes                   |
+| ------------- | ---------------- | ------------------------------------- |
+| Email         | `id@example.com` | `{ "email": "id@example.com" }`       |
+| Username      | `example`        | `{ "preferred_username": "example" }` |
+| Phone         | `+8520001`       | `{ "phone_number": "+8520001" }`      |
+
+OAuth identities also contributes attributes of the user profile, but it is more complicated than the above three identity types. The following sections will discuss the related behavior and configs.
+
+### The built-in oauth identity standard attributes
+
+For each supported oauth provider types, authgear has implemented a built-in standard attribute mappings. You could find the mappings of each provider in the [SSO Providers](/docs/specs/sso-providers.md) document.
+
+### Customizing the oauth identity attributes
+
+As the built-in mappings may not be able to handle all use cases, we support configuring custom mappings on oauth identity attributes. Let's start with the following example config:
+
+```yaml
+identity:
+  oauth:
+    providers:
+      - alias: adfs
+        client_id: exampleclientid
+        type: adfs
+        user_profile_mapping:
+          - from_claim:
+              pointer: "/primary_phone"
+            to_attribute:
```

Comment:
It should be `user_profile_attribute`.

Reply:
actually `user_profile` might be good enough? As we defined `user_profile` = standard attributes + custom attributes elsewhere already.